ang/civicase/activity/actions/services/move-copy-activity-action.service.js

```diff
@@ -3,24 +3,41 @@
 
   module.service('MoveCopyActivityAction', MoveCopyActivityAction);
 
+  /**
+   * Move Copy Activity Action Service
+   *
+   * @param {object} $rootScope rootscope object
+   * @param {object} crmApi service to call civicrm api
+   * @param {object} dialogService service for opening dialog box
+   */
   function MoveCopyActivityAction ($rootScope, crmApi, dialogService) {
     var ts = CRM.ts('civicase');
 
+    /**
+     * Perform the action
+     *
+     * @param {object} $scope scope object
+     * @param {object} action action object
+     */
+    this.doAction = function ($scope, action) {
+      moveCopyActivities($scope.selectedActivities, action.operation, $scope.isSelectAll, $scope.params, $scope.totalCount);
```

Comment:
line too long, consider moving each parameter to its own line.

Also, we rather pass `$scope` and `action` than to pass multiple parameters like this. The problem with multiple parameters is that it's easy to forget the order, and if we need to skip a parameter we need to send `null` or `undefined`.

You can also try passing a single object:

```
moveCopyActivities({
  selectedActivities: $scope.selectedActivities,
  operation: action.operation,
  // ...
});

Reply:
This will involve some time, but given that tomorrow is release and I need to send it for testing today, lets do it later.